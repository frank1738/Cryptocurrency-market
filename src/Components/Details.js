import React from 'react';
import { useParams } from 'react-router-dom';
import { fetchCategory } from '../Redux/Category/categories';
import { useSelector } from 'react-redux';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import DOMPurify from 'dompurify';

const Details = () => {
  const dispatch = useDispatch();
  const params = useParams();
  const id = params.coinId;
  const details = useSelector((state) => state.detail);
  useEffect(() => {
    dispatch(fetchCategory(id));
  }, [dispatch]);
  return (
    <div>
      <div className="detail-container">
        <div className="content">
          <h1 className="detail-name">{details.name}</h1>
        </div>
        <div className="content">
          <div className="rank">
            <span className="rank-btn">Rank # {details.market_cap_rank}</span>
          </div>
          <div className="info">
            <div className="detail-heading">
              {details.image ? (
                <img
                  src={details.image.small}
                  alt="coin"
                  className="detail-img"
                />
              ) : null}

              {details.name ? <p className="det-name">{details.name}</p> : null}
              {details.symbol ? <p>{details.symbol.toUpperCase()}</p> : null}
            </div>
            <div className="detail-price">
              {details.market_data ? (
                <h1>
                  ${details.market_data.current_price.usd.toLocaleString()}
                </h1>
              ) : null}
            </div>
          </div>
        </div>
        <div className="content">
          <table className="table">
            <thead>
              <tr>
                <th>1h</th>
                <th>24h</th>
                <th>7d</th>
                <th>14d</th>
                <th>30d</th>
                <th>1yr</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  {details.market_data?.price_change_percentage_1h_in_currency
                    ? details.market_data.price_change_percentage_1h_in_currency.usd.toFixed(
                        1
                      )
                    : null}
                  %
                </td>
                <td>
                  {details.market_data?.price_change_percentage_24h_in_currency
                    ? details.market_data.price_change_percentage_24h_in_currency.usd.toFixed(
                        1
                      )
                    : null}
                  %
                </td>
                <td>
                  {details.market_data?.price_change_percentage_7d_in_currency
                    ? details.market_data.price_change_percentage_7d_in_currency.usd.toFixed(
                        1
                      )
                    : null}
                  %
                </td>
                <td>
                  {details.market_data?.price_change_percentage_14d_in_currency
                    ? details.market_data.price_change_percentage_14d_in_currency.usd.toFixed(
                        1
                      )
                    : null}
                  %
                </td>
                <td>
                  {details.market_data?.price_change_percentage_30d_in_currency
                    ? details.market_data.price_change_percentage_30d_in_currency.usd.toFixed(
                        1
                      )
                    : null}
                  %
                </td>
                <td>
                  {details.market_data?.price_change_percentage_1y_in_currency
                    ? details.market_data.price_change_percentage_1y_in_currency.usd.toFixed(
                        1
                      )
                    : null}
                  %
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div className="content">
          <div className="stats">
            <div className="left">
              <div className="row first-row">
                <h4>24 Hour Low</h4>
                {details.market_data?.low_24h ? (
                  <p className="data">
                    ${details.market_data.low_24h.usd.toLocaleString()}
                  </p>
                ) : null}
              </div>
              <div className="row ">
                <h4>24 Hour High</h4>
                {details.market_data?.high_24h ? (
                  <p className="data">
                    ${details.market_data.high_24h.usd.toLocaleString()}
                  </p>
                ) : null}
              </div>
            </div>
            <div className="right">
              <div className="row">
                <h4>Market Cap</h4>
                {details.market_data?.market_cap ? (
                  <p className="data">
                    ${details.market_data.market_cap.usd.toLocaleString()}
                  </p>
                ) : null}
              </div>
              <div className="row">
                <h4>Circulating Supply</h4>
                {details.market_data?.circulating_supply ? (
                  <p className="data">
                    ${details.market_data.circulating_supply.toLocaleString()}
                  </p>
                ) : null}
              </div>
            </div>
          </div>
        </div>
        <div className="content">
          <div className="about">
            <h3 className="about-header">About</h3>
            <p
              className="about-description"
              dangerouslySetInnerHTML={{
                __html: DOMPurify.sanitize(
                  details.description ? details.description.en : ''
                ),
              }}
            ></p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Details;
