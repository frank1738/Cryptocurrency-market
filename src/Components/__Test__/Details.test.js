import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Details from '../Details';
import { store } from '../../Redux/configureStore';

describe('Details page test', () => {
  test('should render Details page properly', () => {
    const component = render(
      <Provider store={store}>
        <Details />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
