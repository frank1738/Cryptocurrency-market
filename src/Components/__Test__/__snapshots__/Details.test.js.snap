// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page test should render Details page properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="detail-container"
        >
          <div
            class="content"
          >
            <h1
              class="detail-name"
            />
          </div>
          <div
            class="content"
          >
            <div
              class="rank"
            >
              <span
                class="rank-btn"
              >
                Rank # 
              </span>
            </div>
            <div
              class="info"
            >
              <div
                class="detail-heading"
              />
              <div
                class="detail-price"
              />
            </div>
          </div>
          <div
            class="content"
          >
            <table
              class="table"
            >
              <thead>
                <tr>
                  <th>
                    1h
                  </th>
                  <th>
                    24h
                  </th>
                  <th>
                    7d
                  </th>
                  <th>
                    14d
                  </th>
                  <th>
                    30d
                  </th>
                  <th>
                    1yr
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    %
                  </td>
                  <td>
                    %
                  </td>
                  <td>
                    %
                  </td>
                  <td>
                    %
                  </td>
                  <td>
                    %
                  </td>
                  <td>
                    %
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="content"
          >
            <div
              class="stats"
            >
              <div
                class="left"
              >
                <div
                  class="row first-row"
                >
                  <h4>
                    24 Hour Low
                  </h4>
                </div>
                <div
                  class="row "
                >
                  <h4>
                    24 Hour High
                  </h4>
                </div>
              </div>
              <div
                class="right"
              >
                <div
                  class="row"
                >
                  <h4>
                    Market Cap
                  </h4>
                </div>
                <div
                  class="row"
                >
                  <h4>
                    Circulating Supply
                  </h4>
                </div>
              </div>
            </div>
          </div>
          <div
            class="content"
          >
            <div
              class="about"
            >
              <h3
                class="about-header"
              >
                About
              </h3>
              <p
                class="about-description"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="detail-container"
      >
        <div
          class="content"
        >
          <h1
            class="detail-name"
          />
        </div>
        <div
          class="content"
        >
          <div
            class="rank"
          >
            <span
              class="rank-btn"
            >
              Rank # 
            </span>
          </div>
          <div
            class="info"
          >
            <div
              class="detail-heading"
            />
            <div
              class="detail-price"
            />
          </div>
        </div>
        <div
          class="content"
        >
          <table
            class="table"
          >
            <thead>
              <tr>
                <th>
                  1h
                </th>
                <th>
                  24h
                </th>
                <th>
                  7d
                </th>
                <th>
                  14d
                </th>
                <th>
                  30d
                </th>
                <th>
                  1yr
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  %
                </td>
                <td>
                  %
                </td>
                <td>
                  %
                </td>
                <td>
                  %
                </td>
                <td>
                  %
                </td>
                <td>
                  %
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="content"
        >
          <div
            class="stats"
          >
            <div
              class="left"
            >
              <div
                class="row first-row"
              >
                <h4>
                  24 Hour Low
                </h4>
              </div>
              <div
                class="row "
              >
                <h4>
                  24 Hour High
                </h4>
              </div>
            </div>
            <div
              class="right"
            >
              <div
                class="row"
              >
                <h4>
                  Market Cap
                </h4>
              </div>
              <div
                class="row"
              >
                <h4>
                  Circulating Supply
                </h4>
              </div>
            </div>
          </div>
        </div>
        <div
          class="content"
        >
          <div
            class="about"
          >
            <h3
              class="about-header"
            >
              About
            </h3>
            <p
              class="about-description"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
